### DISCUSSION TITLE
Extensions issues in different Python versions

### DISCUSSION THREAD

**Discussion by dnzbk** [nzbgetcom/nzbget#56] [General]
VideoSort cannot work with python 3.12.x,  because in this version was removed [imp](https://docs.python.org/3.4/library/imp.html#module-imp),  which is a dependency of [bablefish](https://pypi.org/project/babelfish/) which still uses pkg_resources.py.

**Comment by Safihre** [comment]
If you install `setuptools`, it works as this includes `pkg_resources`.
Since `setuptools` is a very common package, it's very likely that users already have it installed.

**Reply by dnzbk** [reply]
Yes, you're right. We've already solved this problem. The latest [9.0](https://github.com/nzbgetcom/Extension-VideoSort/releases/tag/v9.0) version already works fine with Python 3.12

**Reply by Safihre** [reply]
Oh I see you vendor all libraries. That's another way to go of course.

**Reply by dnzbk** [reply]
Yes, we do, but from my point of view it is not the best solution. This is how the project developed initially and we've left it as it is for now. I think that later we will switch to requirements.txt as a more preferable and flexible way for dependency handling.
